ui/app/components/search-select.js

```diff
@@ -0,0 +1,99 @@
+import Component from '@ember/component';
+import { inject as service } from '@ember/service';
+import { task } from 'ember-concurrency';
+import { computed } from '@ember/object';
+
+export default Component.extend({
+  'data-test-component': 'search-select',
+  classNames: ['field', 'search-select'],
+  store: service(),
+  /*
+   * @public
+   * @param Function
+   *
+   * Function called when any of the inputs change
+   * accepts a single param `value` that is the
+   * result of calling `toVal()`.
+   *
+   */
+  onChange: () => {},
+
+  /*
+   * @public
+   * @param String | Array
+   * A comma-separated string or an array of strings.
+   * Defaults to an empty array.
+   *
+   */
+  inputValue: computed(function() {
+    return [];
+  }),
+  selectedOption: null,
+  selectedOptions: null,
+  options: null,
+  shouldUseFallback: false,
+  init() {
+    this._super(...arguments);
+    this.set('selectedOptions', this.inputValue || []);
+  },
+  fetchOptions: task(function*() {
+    for (let modelType of this.models) {
+      yield this.store
+        .adapterFor(modelType)
+        .query(null, { modelName: modelType })
+        .then(resp => {
```

Comment:
If you used the store method ( `let models = this.store.query(modelType, {})` ) for fetching - you'd get back models and not have to do any of this looping / iteration (the serializers will already be doing that).  I'm not quite sure how that would work with the ember-power-select stuff though. Was there a reason to use the adapters directly here?